src/2048-expert/index.js

```diff
@@ -187,6 +189,7 @@ function keyUpLeft() {
   moveLeft();
   generateTwoNewTile();
   addColours();
+  checkLost();
```

Comment:
Why are we calling `checkLost()` after it's already been called as part of `mergeHorizontal()`? It will also be called as part of `generateTwoNewTile()` if required.

Reply:
Sometimes this could happen when the board is completely full but still playable, it will be a minor difference but I'm more intent to encourage the player to not give up all the possibilities for winning the game/ getting a higher score. I think it is more reasonable to check it after every move player makes.

Reply:
I don't think we need the extra call to `checkLost()` though?

Reply:
The loss condition won't have been activated between the `mergeHorizontal/Vertical()` call and this second call afaict.